contracts/StakeVault.sol

```diff
@@ -9,64 +9,226 @@ import { StakeManager } from "./StakeManager.sol";
+     */
     function stake(uint256 _amount, uint256 _time) external onlyOwner {
-        bool success = STAKED_TOKEN.transferFrom(msg.sender, address(this), _amount);
-        if (!success) {
-            revert StakeVault__StakingFailed();
-        }
-        stakeManager.stake(_amount, _time);
+        _stake(_amount, _time, msg.sender);
+    }
 
-        emit Staked(msg.sender, address(this), _amount, _time);
+    /**
+     * @notice Stake tokens from a specified address for a specified time.
+     * @param _amount The amount of tokens to stake.
+     * @param _time The time period to stake for.
+     * @param _from The address from which tokens will be transferred.
+     */
+    function stake(uint256 _amount, uint256 _time, address _from) external onlyOwner {
+        _stake(_amount, _time, _from);
     }
 
+    /**
+     * @notice Extends the lock time of the stake.
+     * @param _time The additional time to lock the stake.
+     */
     function lock(uint256 _time) external onlyOwner {
         stakeManager.lock(_time);
     }
 
+    /**
+     * @notice Unstake a specified amount of tokens and send to the owner.
+     * @param _amount The amount of tokens to unstake.
+     */
     function unstake(uint256 _amount) external onlyOwner {
-        stakeManager.unstake(_amount);
-        bool success = STAKED_TOKEN.transfer(msg.sender, _amount);
-        if (!success) {
-            revert StakeVault__UnstakingFailed();
-        }
+        _unstake(_amount, msg.sender);
     }
 
+    /**
+     * @notice Unstake a specified amount of tokens and send to a destination address.
+     * @param _amount The amount of tokens to unstake.
+     * @param _destination The address to receive the unstaked tokens.
+     */
+    function unstake(uint256 _amount, address _destination) external onlyOwner validDestination(_destination) {
+        _unstake(_amount, _destination);
+    }
+
+    /**
+     * @notice Withdraw tokens from the contract.
+     * @param _token The ERC20 token to withdraw.
+     * @param _amount The amount of tokens to withdraw.
+     */
+    function withdraw(ERC20 _token, uint256 _amount) external onlyOwner {
+        _withdraw(_token, _amount, msg.sender);
+    }
+
+    /**
+     * @notice Withdraw tokens from the contract to a destination address.
+     * @param _token The ERC20 token to withdraw.
+     * @param _amount The amount of tokens to withdraw.
+     * @param _destination The address to receive the tokens.
+     */
+    function withdraw(
+        ERC20 _token,
+        uint256 _amount,
+        address _destination
+    )
+        external
+        onlyOwner
+        validDestination(_destination)
+    {
+        _withdraw(_token, _amount, _destination);
+    }
+
+    /**
+     * @notice Withdraw Ether from the contract to the owner address.
+     * @param _amount The amount of Ether to withdraw.
+     */
+    function withdraw(uint256 _amount) external onlyOwner {
+        _withdraw(_amount, payable(msg.sender));
+    }
```

Comment:
I just realized, there's actually very little point in having this here because this contract cannot receive any eth. There's no `receive()` or `fallback` implementation here. So we might as well drop this.